core/codegen_next/src/attribute/route.rs

```diff
@@ -181,7 +181,7 @@ fn data_expr(ident: &syn::Ident, ty: &syn::Type) -> TokenStream2 {
             Borrowed(Outcome::Success(ref __v)) => {
                 Borrowed(Outcome::Success(::std::borrow::Borrow::borrow(__v)))
             }
-            Borrowed(__o) => Borrowed(__o.map(|_| loop { /* unreachable */ })),
+            Borrowed(__o) => Borrowed(__o.map(|_| loop { unreachable!("Unreachable code"); })),
```

Comment:
```suggestion
            Borrowed(__o) => Borrowed(__o.map(unreachable!("case handled in previous match block"))),
```

Reply:
I don't think this is right -- this looks like it evaluates `unreachable!` before entering `map()`